src/snapred/backend/data/Indexer.py

```diff
@@ -342,68 +303,111 @@ def addIndexEntry(self, entry: IndexEntry):
         Will save at the version on the index entry.
         If the version is invalid, will throw an error and refuse to save.
         """
-        if not self.isValidVersion(entry.version):
-            raise RuntimeError(f"Invalid version {entry.version} on index entry.  Save failed.")
-
+        entry.version = self._flattenVersion(entry.version)
         self.index[entry.version] = entry
         self.writeIndex()
 
     ## RECORD READ / WRITE METHODS ##
 
     def createRecord(self, *, version, **other_arguments):
         record = RECORD_TYPE[self.indexerType](
-            version=self.thisOrNextVersion(version),
+            version=self._flattenVersion(version),
             **other_arguments,
         )
         record.calculationParameters.version = record.version
         return record
 
-    def _determineRecordType(self, version: Optional[int] = None):
-        version = self.thisOrCurrentVersion(version)
+    def _determineRecordType(self, version: int):
```

Comment:
Should `version` be ` Version`?

Reply:
again not, as above.  If its on disk it better be an int.